### PR TITLE
Added cross validation techniques

### LINKED ISSUE #32: Add cross validation techniques
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

I would like to propose adding cross-validation techniques to the drug review project. Currently, the project includes algorithms such as linear regression, logistic regression, perceptron, and decision tree. By incorporating cross-validation, we can obtain more robust evaluations of these algorithms and improve the model's performance.

I suggest using scikit-learn's cross_val_score function and KFold for the cross-validation strategy. This example can serve as a starting point for implementing cross-validation with other algorithms as well.

By incorporating cross-validation, we can better assess the performance of the models, estimate their generalization capabilities, and potentially fine-tune hyperparameters to enhance their accuracy. This addition will make the project more reliable and comprehensive.

### Screenshots

_No response_

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

### PR DESCRIPTION
**Pull request by KaviRana**
# Fixes Issue🛠️

Closes #32 

# Description👨‍💻 

This pull request adds a function for cross validation techniques that can be looped through each model. This will allow us to more accurately evaluate the performance of our models.

The following dependencies are required for this change:

1. numpy
2. scikit-learn

# Type of change📄

- [x] New feature (non-breaking change which adds functionality)

# How this has been tested✅
I have conducted unit test for my function. 
# Checklist✅ 

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have added demonstration in the form of GIF/video file
- [x] I am an Open Source Contributor

# Screenshots/GIF📷
image

### CODE REVIEW

**Comment by Rakesh9100** [comment]
Provide the output screenshot of the successful running of the code without any error @KaviRana

**Comment by KaviRana** [comment]
Function requires the models to be correctly fitted into data. This wont work until and unless the model are working properly and fixing fitting issues is out of scope for my issue.

**Comment by Rakesh9100** [comment]
Every model that are present in the code is working properly. You can see readme file as well as run the code in your local.
You have to provide the output screenshot to check the flawless run of program after your changes. @KaviRana

**Comment by KaviRana** [comment]
ok I am going to figure this out soon.

**Comment by Rakesh9100** [comment]
Pls provide the status @KaviRana and also do resolve the conflicts
